src/biokbase/narrative/tests/generate_test_results.py

```diff
@@ -42,10 +37,37 @@ def get_test_spec(tag, app_id):
     return copy.deepcopy(TEST_SPECS[tag][app_id])
 
 
+def generate_mappings(all_jobs):
```

Comment:
Rather than hard-coding mappings like `RETRIED_JOBS` or `JOBS_BY_CELL_ID`, these are now generated directly from the test data.